src/Microsoft.TemplateEngine.Cli/TemplateSearch/CliTemplateSearchCoordinator.cs

```diff
@@ -224,18 +230,51 @@ internal string PrintableTotalDownloads
                     }
                     else if (TotalDownloads < 1000)
                     {
-                        return "<1k";
+                        return MinimumDownloadCount;
                     }
                     else
                     {
-                        return $"{(TotalDownloads / 1000):N0}k";
+                        return $"{(TotalDownloads / 1000):N0}{ThousandsChar}";
                     }
                 }
             }
 
             internal TemplateGroupTableRow TemplateGroupInfo { get; private set; }
 
             internal long TotalDownloads { get; private set; }
+
+            private class DownloadCountComparer : IComparer<string>
+            {
+                public int Compare(string? x, string? y)
+                {
+                    if (x == y || string.IsNullOrWhiteSpace(x) && string.IsNullOrWhiteSpace(y))
+                    {
+                        return 0;
+                    }
+                    if (string.IsNullOrWhiteSpace(x))
+                    {
+                        return -1;
+                    }
+                    if (string.IsNullOrWhiteSpace(y))
+                    {
+                        return 1;
+                    }
+                    int xInt = 0;
+                    int yInt = 0;
+
+                    if (x != MinimumDownloadCount)
+                    {
+                        int.TryParse(x.Substring(0, x.Length - 1), out xInt);
```

Comment:
Why `-1`? Is it length of the word `k`? If yes, I prefer `ThousandsChar.Length`.

Reply:
`ThousandsChar` is a char. not a string.

Reply:
Ah, that's fair. I actually had the assumption in mind that this would be localized, in which case it could be more than 1 character. If you convert the thousands symbol to a `string` for loc purposes, please update this as well.

Otherwise, maybe a comment could help future devs.
```suggestion
                        // x.Length - 1 : remove last character which is always 'k'.
                        int.TryParse(x.Substring(0, x.Length - 1), out xInt);
```

Reply:
depends on decision made in https://github.com/dotnet/templating/pull/4189#discussion_r766764265

Reply:
not relevant anymore